tests/bluetooth/audio/cap_commander/src/test_distribute_broadcast_code.c

```diff
@@ -0,0 +1,274 @@
+	}
+
+	fixture->broadcast_assistant_cb.recv_state =
+		cap_commander_broadcast_assistant_recv_state_cb;
+	err = bt_bap_broadcast_assistant_register_cb(&fixture->broadcast_assistant_cb);
+	zassert_equal(0, err, "Failed registering broadcast assistant callback functions %d", err);
+}
+
+static void *cap_commander_test_distribute_broadcast_code_setup(void)
+{
+	struct cap_commander_test_distribute_broadcast_code_fixture *fixture;
+
+	fixture = malloc(sizeof(*fixture));
+	zassert_not_null(fixture);
+
+	return fixture;
+}
+
+static void cap_commander_test_distribute_broadcast_code_before(void *f)
+{
+	int err;
+	struct cap_commander_test_distribute_broadcast_code_fixture *fixture = f;
+
+	memset(f, 0, sizeof(struct cap_commander_test_distribute_broadcast_code_fixture));
+	cap_commander_test_distribute_broadcast_code_fixture_init(fixture);
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		err = bt_cap_commander_discover(&fixture->conns[i]);
+		zassert_equal(0, err, "Unexpected return value %d", err);
+	}
+}
+
+static void cap_commander_test_distribute_broadcast_code_after(void *f)
+{
+	struct cap_commander_test_distribute_broadcast_code_fixture *fixture = f;
+
+	bt_cap_commander_unregister_cb(&mock_cap_commander_cb);
+	bt_bap_broadcast_assistant_unregister_cb(&fixture->broadcast_assistant_cb);
+
+	/* We need to cleanup since the CAP commander remembers state */
+	bt_cap_commander_cancel();
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		mock_bt_conn_disconnected(&fixture->conns[i], BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	}
+}
+
+static void cap_commander_test_distribute_broadcast_code_teardown(void *f)
+{
+	free(f);
+}
+
+static void test_distribute_broadcast_code(
+	struct bt_cap_commander_distribute_broadcast_code_param *distribute_broadcast_code_param)
+{
+	int err;
+
+	err = bt_cap_commander_distribute_broadcast_code(distribute_broadcast_code_param);
+	zassert_equal(0, err, "Unexpected return value %d", err);
+
+	zexpect_call_count("bt_cap_commander_cb.distribute_broadcast_code", 1,
+			   mock_cap_commander_distribute_broadcast_code_cb_fake.call_count);
+	zassert_equal_ptr(NULL,
+			  mock_cap_commander_distribute_broadcast_code_cb_fake.arg0_history[0]);
+	zassert_equal(0, mock_cap_commander_distribute_broadcast_code_cb_fake.arg1_history[0]);
+}
+
+ZTEST_SUITE(cap_commander_test_distribute_broadcast_code, NULL,
+	    cap_commander_test_distribute_broadcast_code_setup,
+	    cap_commander_test_distribute_broadcast_code_before,
+	    cap_commander_test_distribute_broadcast_code_after,
+	    cap_commander_test_distribute_broadcast_code_teardown);
+
+ZTEST_F(cap_commander_test_distribute_broadcast_code, test_commander_distribute_broadcast_code)
+{
+	int err;
+
+	err = bt_cap_commander_register_cb(&mock_cap_commander_cb);
+	zassert_equal(0, err, "Unexpected return value %d", err);
+
+	/* we perform a broadcast_reception_start to ensure we get a valid src_id */
+	test_broadcast_reception_start(&fixture->start_param);
```

Comment:
As previously mentioned; we can just initialize the values since there are no peer device involved, right?

Reply:
Yes, I'll do that instead